lib/httpClient.js

```diff
@@ -86,29 +86,43 @@ function Client (opts) {
     this.resData[this.cer].headers = opts
   }
 
+  const emitAndMove = () => {
+    const resp = this.resData[this.cer]
+    this.emit('response', resp.headers.statusCode, resp.bytes, resp.duration)
+    resp.bytes = 0
+    this.cer = this.cer === opts.pipelining - 1 ? 0 : this.cer++
```

Comment:
Can you please avoid getting the value from `opts.pipelining` and set `pipeling` to a variable on top?